metaflow/plugins/pypi/conda_decorator.py

```diff
@@ -0,0 +1,200 @@
+        # Create a symlink to metaflow installed outside the virtual environment
+        self.metaflow_dir = tempfile.TemporaryDirectory(dir="/tmp")
+        os.symlink(
+            os.path.join(get_metaflow_root(), "metaflow"),
+            os.path.join(self.metaflow_dir.name, "metaflow"),
+        )
+
+    def runtime_task_created(
+        self, task_datastore, task_id, split_index, input_paths, is_cloned, ubf_context
+    ):
+        # TODO: Consider recreating the environment if the environment is missing
+        self.interpreter = (
+            self.environment.interpreter(self.step)
+            if not any(
+                decorator.name in ["batch", "kubernetes"]
+                for decorator in next(
+                    step for step in self.flow if step.name == self.step
+                ).decorators
+            )
+            else None
+        )
+
+    def task_pre_step(
+        self,
+        step_name,
+        task_datastore,
+        meta,
+        run_id,
+        task_id,
+        flow,
+        graph,
+        retry_count,
+        max_retries,
+        ubf_context,
+        inputs,
+    ):
+        # Add Python interpreter's parent to the path to ensure that any non-pythonic
+        # dependencies in the virtual environment are visible to the user code.
+        # sys.executable points to the Python interpreter in the virtual environment
+        # since we are already inside the task context.
+        os.environ["PATH"] = os.pathsep.join(
+            filter(
+                None,
+                (
+                    os.path.dirname(os.path.realpath(sys.executable)),
+                    os.environ.get("PATH"),
+                ),
+            )
+        )
+        # TODO: Register metadata
+
+    def runtime_step_cli(
+        self, cli_args, retry_count, max_user_code_retries, ubf_context
+    ):
+        # TODO: Support unbounded for-each
+        # TODO: Check what happens when PYTHONPATH is defined via @environment
+        # Ensure local installation of Metaflow is visible to user code
+        cli_args.env["PYTHONPATH"] = self.metaflow_dir.name
+        # TODO: Verify user site-package isolation behavior
+        #       https://github.com/conda/conda/issues/7707
+        #       Also ref - https://github.com/Netflix/metaflow/pull/178
+        # cli_args.env["PYTHONNOUSERSITE"] = "1"
+        # The executable is already in place for the user code to execute against
+        if self.interpreter:
+            cli_args.entrypoint[0] = self.interpreter
+
+    def runtime_finished(self, exception):
+        self.metaflow_dir.cleanup()
+
+
+class CondaFlowDecorator(FlowDecorator):
+    # TODO: Migrate conda_base keyword to conda for simplicity.
+    name = "conda_base"
+    defaults = {
+        "packages": {},
+        "libraries": {},  # Deprecated! Use packages going forward.
+        "python": None,
+        # TODO: Add support for disabled
+        # TODO: Support `@conda(python='3.10')` before shipping!!
```

Comment:
P0

Reply:
is there an issue with a string version, or only when passing it as a float instead?
This seems to only break with `@conda_base(python=3.10)` which based on the previous docstrings is not the intended use anyway, as the version is expected as a string.

should we validate the supplied parameter and raise an exception in case of non-string version numbers, as the trailing zeros are lost with floats?